### PR TITLE
[CUBRIDMAN-67] dblink to support for remote database access

### PR DESCRIPTION
**Pull request by airnet73**
http://jira.cubrid.org/browse/CUBRIDMAN-67

Purpose
Add cubrid dblink manual (en/ko)

Implementation
N/A

Remarks
N/A

### CODE REVIEW

File: en/sql/dblink.rst [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+cub_gateway
+----------------
+
+cub_broker relays the connection between the application client and the cub_cas_cgw. That is, when an application client requests access, the cub_broker checks the status of the cub_cas_cgw through the shared memory, and then delivers the request to an accessible cub_cas_cgw . It then returns the processing results of the request from the cub_cas_cgw to the application client.
+
+The cub_gateway는 also manages the server load by adjusting the number of cub_cas_cgw (s) in the service pool and monitors and manages the status of the cub_cas_cgw. If the cub_gateway는 delivers the request to cub_cas_cgw but the connection to cub_cas_cgw 1 fails because of an abnormal termination, it sends an error message about the connection failure to the application client and restarts cub_cas_cgw 1. Restarted cub_cas_cgw 1 is now in a normal stand-by mode, and will be reconnected by a new request from a new application client.
```
Comment (beyondykk9) [inline en/sql/dblink.rst:65]: correct error "는"

File: en/sql/dblink.rst [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+공유 메모리
```
Comment (beyondykk9) [inline en/sql/dblink.rst]: correct to "shared memory"

File: en/sql/dblink.rst [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+The following message is returned if the GATEWAY is already running.
+
+::
+
+    cubrid gateway start
+    @ cubrid gateway start
+    ++ cubrid gateway is running.
+
+Stpping GATEWAY
```
Comment (beyondykk9) [inline en/sql/dblink.rst]: correct to "Stopping"

File: en/sql/dblink.rst [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+-------------------------
+
+Enter the command below to stop the GATEWAY.
```
Comment (beyondykk9) [inline en/sql/dblink.rst]:
```suggestion
Enter the below command to stop the GATEWAY.
```

File: en/sql/dblink.rst [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+::
+
+    $ cubrid gateway stop
+    @ cubrid gateway stop
+    ++ cubrid gateway is not running.
+
+Restarting GATEWAY
+---------------------------
+
+Enter the command below to restart the whole GATEWAY.
```
Comment (beyondykk9) [inline en/sql/dblink.rst]:
```suggestion
Enter the below command to restart the whole GATEWAY.
```

File: ko/sql/dblink.rst [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+***********************
+CUBRID DBLink
+***********************
+
+.. _dblink-introduction:
+
+CUBRID DBLink 소개
+==============================================
+
+데이터베이스에서 정보를 조회하다 보면 종종 외부 데이타베이스의 정보 조회가 필요한 경우가 있다. 이렇게 외부 데이터베이스의 정보를 조회하기 위해서 CUBRID DBLink를 이용하면 타 데이터베이스의 정보를 할 수 있다.
```
Comment (beyondykk9) [inline ko/sql/dblink.rst]:
```suggestion
데이터베이스에서 정보를 조회하다 보면 종종 외부 데이타베이스의 정보 조회가 필요한 경우가 있다. 이렇게 외부 데이터베이스의 정보를 조회하기 위해서 CUBRID DBLink를 이용하면 타 데이터베이스의 정보를 조회할 수 있다.

```

File: ko/sql/dblink.rst [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+CUBRID DBLink는 동일 기종인 CUBRID와 이기종인 Oracle, MySQL의 데이타베이스의 정보를 조회할 수 있도록 기능을 제공하고 있다.
+외부 데이타베이스의 정보를 마치 하나의 데이터베이스에서 조회하는 것과 같은 효과를 발휘한다. 단 외부 데이타베이스를 여러 게 설정은 가능 하나, 정보를 조회할 때는 한개의 타 데이타베이스의 정보만 조회가 가능하다.
```
Comment (beyondykk9) [inline ko/sql/dblink.rst]:
```suggestion
외부 데이타베이스의 정보를 마치 하나의 데이터베이스에서 조회하는 것과 같은 효과를 발휘한다. 단 외부 데이타베이스를 여러 개 설정은 가능 하나, 정보를 조회할 때는 한 개의 타 데이타베이스의 정보만 조회가 가능하다.

```

File: ko/sql/dblink.rst [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+
+동일기종의 외부 데이터베이스의 정보를 조회하기 위한 구성도를 보면 Database Server에서 CCI를 이용하여 동일기종의 Brokers에 접속하여 외부 데이터베이스의 정보를 조회할 수 있다.
+
+.. image:: /images/dblink_homo.png
+
+이기종 간의 DBLink 구성도
+-----------------------------
+
+이기종 데이터베이스의 정보를 조회하기 위한 구성도를 보면 GATEWAY를 통해서 이기종 데이터베이스의 정보를 조회할 수 있다. 
+GATWAY는 ODBC(Open DataBase Connectivity)를 이용하고 있다.
```
Comment (beyondykk9) [inline ko/sql/dblink.rst]:
```suggestion
GATEWAY는 ODBC(Open DataBase Connectivity)를 이용하고 있다.
```

File: ko/sql/dblink.rst [hunk 9]
```diff
@@ -0,0 +1,385 @@
...
+
+
+
+
+
+
+제약사항
+==============================================
+
+*   CUBRID DBLink는 utf-8만 지원 한다.
```
Comment (beyondykk9) [inline ko/sql/dblink.rst]:
```suggestion
*   CUBRID Hetergeneous DBLink는 utf-8만 지원 한다.
```

**Review by charischau — APPROVED** [review]

File: en/sql/dblink.rst [hunk 10]
```diff
@@ -0,0 +1,392 @@
...
+
+CUBRID DBLink diagram
+==============================================
+
+CUBRID DBLink supports DBLink between homogeneous and heterogeneous DBLinks.
+
+Homogeneous DBLink diagram
+-----------------------------
+
+If you look at the configuration diagram for inquiring information of homogeneous database, you can use CCI in Database Server to connect to homogeneous brokers and inquire information from external database.
```
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
If you look at the configuration diagram for inquiring information of a homogeneous database, you can use CCI in Database Server to connect to homogeneous brokers and inquire information from an external database.

```

File: en/sql/dblink.rst [hunk 11]
```diff
@@ -0,0 +1,392 @@
...
+
+.. image:: /images/dblink_heter.png
+
+
+.. _gateway:
+
+GATEWAY
+==============================================
+
+A gateway is a middleware that relays to connect to an external database server and is similar to a broker. The gateway connects the CUBRID Database Server to an external server, that is, Oracle/MySQL, retrieves information from the external server and delivers it to the CUBRID Database Server.
```
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
A gateway is a middleware that relays to connect to an external database server and is similar to a broker. The gateway connects the CUBRID Database Server to an external server which is Oracle/MySQL, to retrieve information from the external server and deliver it to the CUBRID Database Server.

```

File: en/sql/dblink.rst [hunk 12]
```diff
@@ -0,0 +1,392 @@
...
+A cubrid system including a gateway has a multi-hierarchical structure including cubrid_gateway, cub_gateway, and cub_cas_cgw as shown in the figure below.
+
+.. image:: /images/gateway.png
+
+cub_cas_cgw
+----------------
+
+cub_cas_cgw(CAS Gateway)는 CUBRID Database Server에서 외부의 Database의 연결을 요청하는 공용 서버 역할을 한다. 또한, cub_cas_cgw는 데이터베이스 서버의 클라이언트로 동작하여 CUBRID Database Server의 요청에 의해 외부 데이터베이스 서버와 연결을 제공한다. 서비스 풀(service pool) 내에서 구동되는 cub_cas_cgw의 개수는 cubrid_gateway.conf 설정 파일에 지정할 수 있으며, cub_gateway에 의해 동적으로 조정된다.
```
Comment (charischau) [inline en/sql/dblink.rst]: should we delete this Korean paragraph since it is the English manual?

File: en/sql/dblink.rst (hunk 3 above)
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
Stopping GATEWAY

```

File: en/sql/dblink.rst [hunk 13]
```diff
@@ -0,0 +1,392 @@
...
+::
+
+    $ cubrid gateway restart
+
+.. _gateway-status:
+
+Checking GATEWAY Status
+--------------------------------
+
+**cubrid gateway status**  The cubrid gateway status utility allows you to check the gateway status such as number of completed jobs and the number of standby jobs by providing various options.
```
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
**cubrid gateway status**  The cubrid gateway status utility allows you to check the gateway status such as the number of completed jobs and the number of standby jobs by providing various options.

```

File: en/sql/dblink.rst [hunk 14]
```diff
@@ -0,0 +1,392 @@
...
++-------------------------------+-------------+------------------------------------------------------------+
+     
+  
+**APPL_SERVER**
+
+    **APPL_SERVER** is the part that sets the application server name of GATEWAY. To connect to an external server, it must be set to CAS_CGW.
+
+**CGW_LINK_SERVER**
+
+    **CGW_LINK_SERVER** should set the name of the external databaseto be used by connecting to CAS_CGW. Currently supported databases are Oracle and MySQL.
```
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
    **CGW_LINK_SERVER** should set the name of the external database to be used by connecting to CAS_CGW. Currently, supported databases are Oracle and MySQL.

```

File: en/sql/dblink.rst [hunk 15]
```diff
@@ -0,0 +1,392 @@
...
+    **CGW_LINK_ODBC_DRIVER_NAME** must set the ODBC Driver name provided by the external database when connecting with CAS_CGW.
+
+.. note::
+    
+        *   For Windows, if the ODBC Driver is installed, the driver name can be found through the ODBC Data Source Manager.
+        *   For Linux, the driver name must be written directly in odbcinit.ini.
+
+**CGW_LINK_CONNECT_URL_PROPERTY**
+
+    **CGW_LINK_CONNECT_URL_PROPERTY** creates a property used in the connection string when connecting CAS_CGW to an external
```
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
    **CGW_LINK_CONNECT_URL_PROPERTY** creates a property used in the connection string when connecting CAS_CGW to an external database. 

```

File: en/sql/dblink.rst [hunk 16]
```diff
@@ -0,0 +1,392 @@
...
+
+.. note::
+    
+        For reference, in the ondbcinst.ini setting, the driver names are MySQL ODBC 8.0 Unicode Driver and Oracle 12c ODBC driver, respectively.
+
+
+How to use Cubrid DBLink
+==============================================
+
+If you set up homogeneous brokers and heterogeneous gateways, let's look at how to write Query statements to inquire about database information.
```
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
In the case of setting up homogeneous brokers and heterogeneous gateways, let's look at how to write Query statements to inquire about database information.

```
or 
```suggestion
If you want to set up homogeneous brokers and heterogeneous gateways, let's look at how to write Query statements to inquire about database information.

```

File: en/sql/dblink.rst [hunk 17]
```diff
@@ -0,0 +1,392 @@
...
+
+First, how to query information from other databases by writing DBLINK syntax in the FROM clause
+The Query statement below is a Query statement that inquires the remote_t table information of another database of IP 192.168.0.1.
+
+::
+    
+	SELECT * FROM DBLINK (192.168.0.1:53000:demodb:user:password:','SELECT col1, col2 FROM remote_t') AS t(col1 int, col2 varchar(32));
+
+
+Second, if you look at the above DBLINK Query, information for accessing other databases is the most basic information. Therefore, there is a risk that user information (id, password) may be exposed to the outside and the inconvenience of having to write each time a Query is written.
```
Comment (charischau) [inline en/sql/dblink.rst]:
```suggestion
Second, if you look at the above DBLINK Query, information for accessing other databases is the most basic information. Therefore, there is a risk that user information (id, password) may be exposed to the outside and there will be an inconvenience of having to write each time a Query is written.

```

File: en/sql/dblink.rst (hunk 12 above)
Comment (airnet73) [reply en/sql/dblink.rst]: thank you I will delete it.

File: en/sql/dblink.rst (hunk 1 above)
Comment (airnet73) [reply en/sql/dblink.rst:65]: I will change it.

File: en/sql/dblink.rst (hunk 2 above)
Comment (airnet73) [reply en/sql/dblink.rst]: I will change it.

File: en/sql/dblink.rst (hunk 3 above)
Comment (airnet73) [reply en/sql/dblink.rst]: I will change it.

Comment (airnet73) [reply en/sql/dblink.rst]: I will change it.

**Review by beyondykk9 — APPROVED** [review]

**Review by charischau — APPROVED** [review]

File: ko/sql/dblink.rst [hunk 18]
```diff
@@ -0,0 +1,387 @@
...
+GATEWAY 구동
+---------------
+
+GATEWAY를 구동하기 위하여 다음과 같이 입력한다. 
+
+::
+
+    $ cubrid gateway start
+    @ cubrid gateway start
+    ++ cubrid gateway start: success
```
Comment (kisoo-han) [inline ko/sql/dblink.rst]: 위에 입력한다고 기술했으니, 이 Line들은 삭제가 바람직해보임

File: ko/sql/dblink.rst [hunk 19]
```diff
@@ -0,0 +1,387 @@
...
+GATEWAY 종료
+---------------
+
+GATEWAY가를 종료하기 위하여 다음과 같이 입력한다. 
+
+::
+
+    $ cubrid gateway stop
+    @ cubrid gateway stop
+    ++ cubrid gateway stop: success
```
Comment (kisoo-han) [inline ko/sql/dblink.rst]: 이 Line도 삭제가 바람직

File: en/sql/dblink.rst [hunk 20]
```diff
@@ -0,0 +1,400 @@
...
+Start GATEWAY
+-----------------------
+
+Enter the command below to start the GATEWAY.
+
+::
+
+    $ cubrid gateway start
+    @ cubrid gateway start
+    ++ cubrid gateway start: success
```
Comment (kisoo-han) [inline en/sql/dblink.rst]: Please delete above lines

File: en/sql/dblink.rst [hunk 21]
```diff
@@ -0,0 +1,400 @@
...
+
+-------------------------
+
+Enter the below command to stop the GATEWAY.
+
+::
+
+    $ cubrid gateway stop
+    @ cubrid gateway stop
+    ++ cubrid gateway stop: success
```
Comment (kisoo-han) [inline en/sql/dblink.rst]: delete lines

**Review by kisoo-han — APPROVED** [review]

**Pull request merged**